airflow/www/views.py

```diff
@@ -882,6 +892,18 @@ def index(self):
                 for name, in dagtags
             ]
 
+            owner_links_dict = {}
+            owner_links = session.query(DagOwnerAttributes).all()
+            # The structure we are going for is:
+            # {dag1: {owner1: link1, owner2: link2}, dag2: {owner1: link1}}
+            for owner_link_pair in owner_links:
+                owner_link_pair_dict = owner_link_pair.as_dict()
+                for dag in owner_link_pair_dict:
+                    if dag in owner_links_dict:
+                        owner_links_dict[dag].update(owner_link_pair_dict[dag])
+                    else:
+                        owner_links_dict.update(owner_link_pair_dict)
```

Comment:
I wonder if this logic belongs to a method on `DagOwnerAttributes` instead, say

```python
@classmethod
def get_all(cls, session) -> Dict[str, Dict[str, str]]:
    dag_links = collections.defaultdict(dict)
    for obj in session.query(cls):
        dag_links[obj.dag_id].update({obj.owner: obj.link})
    return dag_links
```

(does this also mean `as_dict` isn’t needed anymore?)

Reply:
I changed it and deleted the `as_dict`. Your suggestion is indeed more elegant and better :) Thanks!